lib/core/job_cleanup.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core'
+require 'utility/logger'
+
+module Core
+  class JobCleanUp
+    class << self
+      def execute(connector_id = nil)
+        process_orphaned_jobs
+        process_stuck_jobs(connector_id)
+      end
+
+      private
+
+      def process_orphaned_jobs
+        Utility::Logger.info('Start cleaning up orphaned jobs...')
+        orphaned_jobs = ConnectorJob.orphaned_jobs
+        if orphaned_jobs.empty?
+          Utility::Logger.info('No orphaned jobs found. Skipping...')
+          return
+        end
+
+        # delete content indicies in case they are re-created by sync job
+        content_indices = orphaned_jobs.map(&:index_name).compact.uniq
+        ElasticConnectorActions.delete_indices(content_indices) if content_indices.any?
```

Comment:
I'm confused - I don't expect a class called "JobCleanUp" to delete content indices. Maybe should be renamed to something like "OrhanedIndicesCleanUp" and done separately from Job clean up?

Reply:
This was discussed [here](https://elastic.slack.com/archives/C03GAPTQ2JF/p1667984997829019)

If a connector is deleted during job execution, the content index may be recreated, before the connector service stops execution.

That's why we want to make sure the content indices of orphaned jobs are deleted as well.